komorebi-core/src/direction.rs

```diff
@@ -19,10 +19,10 @@ pub trait Direction {
         idx: usize,
         count: usize,
     ) -> bool;
-    fn up_index(&self, idx: usize) -> usize;
-    fn down_index(&self, idx: usize) -> usize;
-    fn left_index(&self, idx: usize) -> usize;
-    fn right_index(&self, idx: usize) -> usize;
+    fn up_index(&self, op_direction: OperationDirection, idx: usize, count: usize) -> usize;
```

Comment:
Since this is ignored when implemented for `CustomLayout` let's make this an `Option<OperationDirection>` so that it's clear for different structs that implement `Direction` whether this is operated on or not